app/models/concerns/statusable.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+module Statusable
```

Comment:
@bwatson78  - thanks so much for that context. Do we have existing baselining set up to test against? (I'll also poke around and see if I can find it.

Reply:
So, it looks like the maximum page loads are still under a second in my local development environment, and I assume they would be faster on the production servers (see table below from JMeter).

I also think it would be worthwhile to think about how to manage when the Alma api is, for whatever reason, unavailable or slow to respond, because right now I believe the page doesn't render if the app doesn't get a response from Alma.

|timeStamp    |elapsed|label                           |responseCode|responseMessage|threadName                       |dataType|success|failureMessage|bytes |sentBytes|grpThreads|allThreads|URL                                                       |Latency|IdleTime|Connect|
|-------------|-------|--------------------------------|------------|---------------|---------------------------------|--------|-------|--------------|------|---------|----------|----------|----------------------------------------------------------|-------|--------|-------|
|1622638963242|494    |Homepage                        |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57751 |116      |1         |1         |http://localhost:3000/                                    |493    |0       |1      |
|1622638963736|1410   |Search Results - Music          |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |105416|616      |1         |1         |http://localhost:3000/?utf8=✓&search_field=keyword&q=Music|1410   |0       |0      |
|1622638965146|1496   |Load Show Page - Minute Metadata|200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57131 |658      |1         |1         |http://localhost:3000/catalog/9936957513502486            |1496   |0       |0      |
|1622638966643|1076   |Load Show Page - Large Metadata |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57730 |658      |1         |1         |http://localhost:3000/catalog/9937177765002486            |1076   |0       |0      |
|1622638967719|457    |Homepage                        |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57813 |634      |1         |1         |http://localhost:3000/                                    |456    |0       |0      |
|1622638968176|1335   |Search Results - Music          |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |105416|678      |1         |1         |http://localhost:3000/?utf8=✓&search_field=keyword&q=Music|1335   |0       |0      |
|1622638969512|1264   |Load Show Page - Minute Metadata|200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57131 |658      |1         |1         |http://localhost:3000/catalog/9936957513502486            |1264   |0       |0      |
|1622638970776|2186   |Load Show Page - Large Metadata |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57730 |658      |1         |1         |http://localhost:3000/catalog/9937177765002486            |2186   |0       |0      |
|1622638972962|562    |Homepage                        |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57813 |634      |1         |1         |http://localhost:3000/                                    |562    |0       |0      |
|1622638973526|1384   |Search Results - Music          |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |105416|678      |1         |1         |http://localhost:3000/?utf8=✓&search_field=keyword&q=Music|1383   |0       |0      |
|1622638974910|2769   |Load Show Page - Minute Metadata|200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57131 |658      |1         |1         |http://localhost:3000/catalog/9936957513502486            |2769   |0       |0      |
|1622638977680|1423   |Load Show Page - Large Metadata |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57730 |658      |1         |1         |http://localhost:3000/catalog/9937177765002486            |1423   |0       |0      |
|1622638979104|462    |Homepage                        |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57813 |634      |1         |1         |http://localhost:3000/                                    |462    |0       |0      |
|1622638979567|1323   |Search Results - Music          |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |105416|678      |1         |1         |http://localhost:3000/?utf8=✓&search_field=keyword&q=Music|1323   |0       |0      |
|1622638980891|1389   |Load Show Page - Minute Metadata|200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57131 |658      |1         |1         |http://localhost:3000/catalog/9936957513502486            |1389   |0       |0      |
|1622638982281|1200   |Load Show Page - Large Metadata |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57730 |658      |1         |1         |http://localhost:3000/catalog/9937177765002486            |1200   |0       |0      |
|1622638983481|538    |Homepage                        |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57813 |634      |1         |1         |http://localhost:3000/                                    |538    |0       |0      |
|1622638984020|1535   |Search Results - Music          |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |105416|678      |1         |1         |http://localhost:3000/?utf8=✓&search_field=keyword&q=Music|1535   |0       |0      |
|1622638985555|1166   |Load Show Page - Minute Metadata|200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57131 |658      |1         |1         |http://localhost:3000/catalog/9936957513502486            |1166   |0       |0      |
|1622638986722|1237   |Load Show Page - Large Metadata |200         |OK             |Navigating Blacklight Catalog 1-1|text    |true   |              |57730 |658      |1         |1         |http://localhost:3000/catalog/9937177765002486            |1237   |0       |0      |

Reply:
That's great. Bringing this over to the Ruby side would help everyone, so I'll retract my worry.